atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampUtils.java

```diff
@@ -0,0 +1,244 @@
+import com.google.protobuf.ByteString;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.table.description.ColumnMetadataDescription;
+import com.palantir.atlasdb.table.description.ColumnValueDescription;
+import com.palantir.atlasdb.table.description.NameComponentDescription;
+import com.palantir.atlasdb.table.description.NameMetadataDescription;
+import com.palantir.atlasdb.table.description.NamedColumnDescription;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.util.Pair;
+
+public final class CassandraTimestampUtils {
+    public static final TableMetadata TIMESTAMP_TABLE_METADATA = new TableMetadata(
+            NameMetadataDescription.create(ImmutableList.of(
+                    new NameComponentDescription("timestamp_name", ValueType.STRING))),
+            new ColumnMetadataDescription(ImmutableList.of(
+                    new NamedColumnDescription(
+                            CassandraTimestampUtils.ROW_AND_COLUMN_NAME,
+                            "current_max_ts",
+                            ColumnValueDescription.forType(ValueType.FIXED_LONG)))),
+            ConflictHandler.IGNORE_ALL);
+
+    public static final String ROW_AND_COLUMN_NAME = "ts";
+    public static final String BACKUP_COLUMN_NAME = "oldTs";
+    public static final ByteString INVALIDATED_VALUE = ByteString.copyFrom(new byte[] {0});
+    public static final long INITIAL_VALUE = 10000;
+
+    private static final long CASSANDRA_TIMESTAMP = -1;
+    private static final String ROW_AND_COLUMN_NAME_HEX_STRING = encodeCassandraHexString(ROW_AND_COLUMN_NAME);
+    private static final ByteString CQL_SUCCESS = ByteString.copyFrom(new byte[] {1});
+
+    @VisibleForTesting
+    static final String APPLIED_COLUMN = "[applied]";
+    static final String COLUMN_NAME_COLUMN = "column1";
+    static final String VALUE_COLUMN = "value";
+
+    private CassandraTimestampUtils() {
+        // utility class
+    }
+
+    public static ByteBuffer constructCheckAndSetMultipleQuery(Map<String, Pair<byte[], byte[]>> checkAndSetRequest) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("BEGIN UNLOGGED BATCH\n"); // Safe, because all updates are on the same partition key
+
+        // Safe, because ordering does not apply in batches
+        checkAndSetRequest.entrySet().forEach(entry -> {
+            String columnName = entry.getKey();
+            byte[] expected = entry.getValue().getLhSide();
+            byte[] target = entry.getValue().getRhSide();
+            builder.append(constructCheckAndSetQuery(columnName, expected, target));
+        });
+        builder.append("APPLY BATCH;");
+        return toByteBuffer(builder.toString());
+    }
+
+    public static boolean isValidTimestampData(byte[] data) {
```

Comment:
Ah, this was initially a private static method in `CassandraTimestampBackupRunner`. I'm going to add a bunch of tests.